v2/pkg/contextargs/contextargs.go

```diff
@@ -0,0 +1,19 @@
+package contextargs
+
+import (
+	"net/http/cookiejar"
+	"sync"
+)
+
+type Context struct {
+	Input     string
+	CookieJar *cookiejar.Jar
+
+	// Access to Args must use lock strategies to prevent data races
+	*sync.RWMutex
+	Args map[string]interface{}
+}
+
+func New() Context {
+	return Context{Args: make(map[string]interface{}), RWMutex: &sync.RWMutex{}}
```

Comment:
Since this map is created each time, won't the read be done for arguments even when we are not sharing values? Ideally the map should only be created and initialized if we have values to share and not all the time. Similar to the mutex

Reply:
good suggestion, now the initialization happens at runtime at the first set operation